### DISCUSSION TITLE
Door / Window Sensor MS200

### DISCUSSION THREAD

**Discussion by webwude** [krahabb/meross_lan#281] [General]
Hi.
There is a new product. MS 200 Door / Window Sensor.
https://www.meross.com/en-gc/smart-sensor/smart-door-and-window-sensor/138

It is somehow recognized by merossLan (as it is connected to the Hub), but there are no readings about open or closed windows / door / whatever.
As always it would be nice to have it enabled someday in merossLAN :) in the app, I can see certain readings - and enable push notifications (see attachment).

![Screenshot_20230420-114016](https://user-images.githubusercontent.com/7056632/233327489-a7050eef-ea28-460a-be1c-c6dfeddc27e6.png)
![Screenshot_20230420-113957](https://user-images.githubusercontent.com/7056632/233327586-4766140e-aa19-4519-8553-bd0f1be63ab9.png)

**Comment by krahabb** [comment]
Hello @webwude,
Thank you for submitting this!
In order to proceed to a better identification I would need a [diagnostic/trace](https://github.com/krahabb/meross_lan/wiki/Diagnostics#download-diagnostics) for the hub hosting the device.

While tracing is in progress you could try operate the device from the Meross app so the states polled by `meross_lan` get recorded.
Let at least 30 seconds between each command set so the polling period can kick in and 'snapshot' the device status.  

This new device will likely use unknown commands so the details of gathered informations might lack a lot of stuff but that is a start anyway.

**Comment by webwude** [comment]
Attached you find a diagnostic trace.  I hope it is sufficient - please give me a short feedback - otherwise I will give it another try.

A screenshot when I opened and closed the window as well

![Screenshot_20230420-154942-1](https://user-images.githubusercontent.com/7056632/233992145-02fff25f-86fb-4f11-a42f-e76b2f729a66.png)
[msh300hk-1681998143.csv](https://github.com/krahabb/meross_lan/files/11310273/msh300hk-1681998143.csv)

**Reply by krahabb** [reply]
Awesome!
That's enough..with a bit more of euristics in `meross_lan` it could have worked 'out-of-the-box'
I'm going to implement support for this device too

**Reply by webwude** [reply]
Thx. Please keep me posted as soon as it is available. Do I have to remove the device once to get it proberly recognized or should it work on the fly?

**Reply by krahabb** [reply]
It should work.
Beware the latest version (4.0.0) could raise a 'ban' on your meross account if you configure a Meross cloud profile as per instructions. You should not configure any (even if your devices are Meross-bound I suggest you to wait until I better understand what is going on with the issue #285 )
So, even if the discovery shows you a cloud profile to configure don't do that (just ignore). The 4.0.0 will keep working good (should) with all the new features (ms200 included) except the cloud integration

**Reply by webwude** [reply]
Good news: after updating to 4.0.1 now the device has the attribute window - in state closed. 
Bad news: even after opening the window the status does not change - it changes inside the meross app, but in HA the status is unchanged.

Please reply, if I should support you in any way.

**Reply by krahabb** [reply]
Hello @webwude,
The status of the sensor should be updated every poll cycle so it might take a while (30 sec) before you see the change in HA

If it still doesn't work this way please upload (another) trace so I can inspect what's wrong: I've tested the protocol decoding with your previous trace and it was correctly working but maybe there's some other issue in the release (4.0.1)

The code is now having a lot of refactor in order to to ensure the cloud connection feature of the release works fine with the minimum risk of ban and there's a lot going on so , maybe, I left some bug here or there

**Reply by webwude** [reply]
> Hello @webwude, The status of the sensor should be updated every poll cycle so it might take a while (30 sec) before you see the change in HA
> 
> If it still doesn't work this way please upload (another) trace so I can inspect what's wrong: I've tested the protocol decoding with your previous trace and it was correctly working but maybe there's some other issue in the release (4.0.1)
> 
> The code is now having a lot of refactor in order to to ensure the cloud connection feature of the release works fine with the minimum risk of ban and there's a lot going on so , maybe, I left some bug here or there

I just tried it - and it was the other way around: opening the window was recognized, but closing not - it is still shown as open as we speak.

Attached the trace - and the screenshot from the app with the timestamps of opening and closing. I will give it another try right away. But first I have to set back the state manually to closed.

Thx for cheking.
Best, w.

![Screenshot_20230514-160731](https://github.com/krahabb/meross_lan/assets/7056632/9dc72960-dee6-4655-8249-f1126c3c25e7)


[msh300hk-1684073221.csv](https://github.com/krahabb/meross_lan/files/11472141/msh300hk-1684073221.csv)

**Comment by webwude** [comment]
Ok, another trace - twice open, twice closed. Notification via Meross App received, but no change in HA.

Screenshot with timings and trace attached.

![Screenshot_20230514-161845](https://github.com/krahabb/meross_lan/assets/7056632/69527cdc-ec01-404b-a5bb-c5454588c9fa)
[msh300hk-1684073894.csv](https://github.com/krahabb/meross_lan/files/11472168/msh300hk-1684073894.csv)

**Comment by krahabb** [comment]
I guess I've found the bug: it appears your hub is receiving instant cloud updates (vie Meross MQTT) and there's an issue in this scenario.
These are well presented in the log (I'm happy with that!) but the code is not 'prepared' to receive those cloud updates in full...
new version should fix this for the good!
If my guess is right you could try set the hub configuration in 'HTTP only' and it should work (with status polling every 30 seconds) until the next release

**Reply by webwude** [reply]
Thank you! Works like a charm!
